### DISCUSSION TITLE
This Serverless Function has crashed

### DISCUSSION THREAD

**Discussion by anis-king-911** [vercel/community#4843] [Help] [answered]
### Summary

Hi.
ok so when i was trying a new project i fell in some issues and solved them successfully but this one was burning my mind for the last 2 hours so i came looking for a solution
i was trying to deploy an expressjs app using this code !

```
import express from 'express';
import path from 'path';
import url from 'url';

const app = express();
const port = 3000;

const __filename = url.fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

app.use("/assets", express.static(path.resolve(__dirname, 'public', 'assets')));

app.get('/', (req, res) => {
  res.sendFile(path.resolve(__dirname, 'public', 'index.html'));
});

app.get('/manga', (req, res) => {
  res.sendFile(path.resolve(__dirname, 'public', 'manga.html'));
});

/*
app.get("/*", (req, res) => {
  res.sendFile(path.resolve(__dirname, "frontend", "index.html"));
});
*/

app.listen(port, () => console.log(`http://localhost:${port}/`));

```

### Steps to Reproduce

and the result was ( Serverless Function has crashed )
any idea how to fix it ?

**Comment by amyegan** [comment]
Do you see more details in the runtime logs?

https://vercel.com/docs/observability/runtime-logs

**Reply by anis-king-911** [reply]
It shows this as an error 
```
Uncaught Exception 	{"errorType":"ReferenceError","errorMessage":"require is not defined in ES module scope, you can use import instead\nThis file is being treated as an ES module because it has a '.js' file extension and '/var/task/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","stack":["ReferenceError: require is not defined in ES module scope, you can use import instead","This file is being treated as an ES module because it has a '.js' file extension and '/var/task/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","    at file:///var/task/___now_launcher.js:1:18","    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)"]}
INIT_REPORT Init Duration: 202.95 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
Uncaught Exception 	{"errorType":"ReferenceError","errorMessage":"require is not defined in ES module scope, you can use import instead\nThis file is being treated as an ES module because it has a '.js' file extension and '/var/task/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","stack":["ReferenceError: require is not defined in ES module scope, you can use import instead","This file is being treated as an ES module because it has a '.js' file extension and '/var/task/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","    at file:///var/task/___now_launcher.js:1:18","    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)"]}
INIT_REPORT Init Duration: 218.92 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Unknown application error occurred
Runtime.Unknown
```

With a status code of 500
––Note––––––––––––
The code runs perfectly fine locally !

**Reply by amyegan** [reply]
I built an example repo based on our guide about using Express.js with Vercel, but I wasn't able to reproduce the error you reported. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use to help with debugging?

- My repo: https://github.com/amyegan/express-vercel
- Guide: https://vercel.com/guides/using-express-with-vercel

**Reply by anis-king-911** [reply, marked as answer]
finally found the issue and solved it, it was in the vercel config file
`
"builds":[
      {
          "src":"./index.js",
          "use":"@now/node"
      }
  ]
`
i just needed to change `@now/node` to `@vercel/node` ,
thank you !!

**Comment by ChKarishan** [comment]
I'm also getting same error while deploying an express app on vercel

**Comment by ChKarishan** [comment]
I sorted it out, it was something in vercel.js file configuration

Thanks btw 😊

Regards
Karishan

**Reply by anis-king-911** [reply]
when a web developer or a programer finds a solution, for an issue that many others have, its better for hem and the society to he shares the solution

**Reply by ChKarishan** [reply]
Yeah buddy, I was busy with work load, and wanted to make sure I post the correct solution (by reading vercel guides). 
Tho looks like the vercel config file changes you posted is the solution, but I will comeback if there's something else.

**Reply by doncarlo5** [reply]
I have the same issue. Here is my vercel.json
`{
  "version": 2,
  "builds": [{ "src": "app.js", "use": "@vercel/node" }],
  "routes": [{ "src": "(.*)", "dest": "app.js" }]
}
`
<img width="524" alt="Screenshot 2024-04-23 at 13 45 14" src="https://github.com/vercel/community/assets/125746558/eaf592ac-ab0e-4d00-959d-6c3231d0fa70">



Here is the logs : 
[logs_result.json](https://github.com/vercel/community/files/15076225/logs_result.json)

Any idea why ?
